### DISCUSSION TITLE
Topology Optimization in an empty space without any wall

### DISCUSSION THREAD

**Discussion by chairmanmao256** [mdolab/dafoam#227] [General]
Hi! I'd like to carry out a Topology Optimization in an empty space without any wall. I've run a case of which the computational domain is rectangular and  `inout`  is used for boundary condition. However, DAFoam fails during mesh check before evaluating the objective function. All quantities related to mesh check (bounding box,  min volume and so on) are `nan`. I checked `pyDAFoam.py` and it says that the `designSurface` must be a wall. Is there a way to do Topology Optimization in a computational domain with no `wall` boundary condition imposed? Or do I have to include at least one `wall` boundary for any optimization even if the design variable is a field?

Thank you!!!

**Comment by friedenhe** [comment]
Yes, you have to have at least one wall patch and set it as the design surface. If you do not have a wall, you have to hack pyDAFoam. This line: https://github.com/mdolab/dafoam/blob/e49c73665644966bb63eee055f773cabd71076cc/dafoam/pyDAFoam.py#L3083 You need to change it to: 

`if bc["type"] == "wall" or bc["type"] == "slip" or bc["type"] == "cyclic" or bc["type"] == "patch":`

**Reply by chairmanmao256** [reply]
Thanks a lot!!! I'll read the related code in `pyDAFoam.py` and give it a try!
